blog/2020-09-15-datadog-apm-in-azure-service-fabric.md

```diff
@@ -0,0 +1,193 @@
+
+### Setup an administrator for the script
+
+To install the .NET Tracer in a service, we include some scripts to be deployed into the output directory and executed.
+
+The install script requires machine administrator permissions.
+Add the *SetupAdminUser* to the *Principals* section in the `ApplicationManifest.xml`. If the *Principals* section is missing, add it:
+
+```
+  <Principals>
+    <Users>
+      <User Name="SetupAdminUser">
+        <MemberOf>
+          <SystemGroup Name="Administrators" />
+        </MemberOf>
+      </User>
+    </Users>
+  </Principals>
+```
+
+### Execute the script and opt-in
+
+*This section should be repeated for each service within an application that you would like to monitor:*
+
+---
+
+In your `ApplicationManifest.xml`, within the *ServiceManifestImport* section of the service, give the Setup script *SetupAdminUser* as the executing user.
+
+```diff
+  <ServiceManifestImport>
+    <ServiceManifestRef ServiceManifestName="ServiceThatDeploysDatadogTracerPkg" ServiceManifestVersion="1.0.0" />
+    <ConfigOverrides />
++   <Policies>
++     <RunAsPolicy CodePackageRef="Code" UserRef="SetupAdminUser" EntryPointType="Setup" />
++   </Policies>
+  </ServiceManifestImport>
+```  
+
+Within the `ServiceManifest.xml` of the service you want to monitor, add the reference to the install script.
+Then, add the envrionment variables to opt into profiling.
+
+```diff
+  <CodePackage Name="Code" Version="1.0.0">
++   <SetupEntryPoint>
++     <ExeHost>
++       <Program>DatadogInstall.bat</Program>
++       <WorkingFolder>CodePackage</WorkingFolder>
++     </ExeHost>
++   </SetupEntryPoint>
+    <EntryPoint>
+      <ExeHost>
+        <Program>ServiceThatDeploysDatadogTracer.exe</Program>
+        <WorkingFolder>CodePackage</WorkingFolder>
+      </ExeHost>
+    </EntryPoint>
++   <EnvironmentVariables>
++     <EnvironmentVariable Name="CORE_CLR_ENABLE_PROFILING" Value="1" />
++     <EnvironmentVariable Name="COR_ENABLE_PROFILING" Value="1" />
++   </EnvironmentVariables>
+  </CodePackage>
+```
+
+Include the `DatadogInstall.bat` and `DatadogInstall.ps1` scripts in the project you would like to monitor.
+Copy the scripts from the following two locations:
+ - [Batch Script](https://github.com/DataDog/azureservicefabric-dotnet-tracing-sample/blob/master/VotingWeb/DatadogInstall.bat)
+ - [Powershell Script](https://github.com/DataDog/azureservicefabric-dotnet-tracing-sample/blob/master/VotingWeb/DatadogInstall.ps1)
+
+Configure both files to be copied to the output directory.
+
+![Copy to output directory](https://user-images.githubusercontent.com/1801443/93110062-d05a9480-f682-11ea-8fb4-7b266f576f68.png)
+
+
+Deploy your application to the cluster, generate some load against it, and you should start seeing your infrastructure and traces coming through to your Datadog account.
+
```

Comment:
...to your Datadog account. For example:

- View [distributed traces](https://app.datadoghq.com/apm/traces) in the Datadog portal.
- View a [host map](https://app.datadoghq.com/infrastructure/map) in the Datadog portal.